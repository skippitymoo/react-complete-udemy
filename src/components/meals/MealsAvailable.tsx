import { JSX } from 'react';
import { Meal } from '../menu.types';
import { MealItem } from './MealItem';

const DUMMY_MEALS: Meal[] = [
  {
    id: 'm1',
    name: 'Sushi',
    description: 'Finest fish and veggies',
    price: 22.99,
  },
  {
    id: 'm2',
    name: 'Schnitzel',
    description: 'A german specialty!',
    price: 16.5,
  },
  {
    id: 'm3',
    name: 'Barbecue Burger',
    description: 'American, raw, meaty',
    price: 12.99,
  },
  {
    id: 'm4',
    name: 'Green Bowl',
    description: 'Healthy...and green...',
    price: 18.99,
  },
];

export const MealsAvailable = (): JSX.Element => {
  return (
    <section className='meals-available'>
      <ul>
        {DUMMY_MEALS.map((meal: Meal) => (
          <li key={meal.id}>
            <MealItem>
              <MealItem.Details
                name={meal.name}
                description={meal.description}
                price={meal.price}
              />
              <MealItem.AddToBasket onAddToBasket={() => {}} />
            </MealItem>
          </li>
        ))}
      </ul>
    </section>
  );
};
