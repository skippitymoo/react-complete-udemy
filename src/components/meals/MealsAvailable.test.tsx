import { render } from '@testing-library/react';
import { MealsAvailable } from './MealsAvailable';
import { axe, toHaveNoViolations } from 'jest-axe';

expect.extend(toHaveNoViolations);

describe('MealsAvailable', () => {
  it('should pass axe', async () => {
    const { container } = render(<MealsAvailable />);
    expect(await axe(container)).toHaveNoViolations();
  });

  it('should render as expected', async () => {
    const { container } = render(<MealsAvailable />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
