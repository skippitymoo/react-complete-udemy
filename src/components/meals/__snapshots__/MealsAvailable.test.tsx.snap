// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealsAvailable should render as expected 1`] = `
<section
  class="meals-available"
>
  <ul>
    <li>
      <div
        class="meal-item"
      >
        <div
          class="meal-item__details"
        >
          <h3
            class="meal-item__title"
          >
            Sushi
          </h3>
          <div
            class="meal-item__description"
          >
            Finest fish and veggies
          </div>
          <div
            class="meal-item__price"
          >
            £22.99
          </div>
        </div>
        <form
          class="meal-item__add"
          name="Add meal item"
        >
          <div
            class="form__group"
          >
            <label
              class="form__label"
              for=":r4:"
            >
              Amount
            </label>
            <input
              class="form__input "
              id=":r4:"
              type="number"
              value="1"
            />
          </div>
          <button
            class="btn btn--primary btn--long "
          >
            + Add
          </button>
        </form>
      </div>
    </li>
    <li>
      <div
        class="meal-item"
      >
        <div
          class="meal-item__details"
        >
          <h3
            class="meal-item__title"
          >
            Schnitzel
          </h3>
          <div
            class="meal-item__description"
          >
            A german specialty!
          </div>
          <div
            class="meal-item__price"
          >
            £16.50
          </div>
        </div>
        <form
          class="meal-item__add"
          name="Add meal item"
        >
          <div
            class="form__group"
          >
            <label
              class="form__label"
              for=":r5:"
            >
              Amount
            </label>
            <input
              class="form__input "
              id=":r5:"
              type="number"
              value="1"
            />
          </div>
          <button
            class="btn btn--primary btn--long "
          >
            + Add
          </button>
        </form>
      </div>
    </li>
    <li>
      <div
        class="meal-item"
      >
        <div
          class="meal-item__details"
        >
          <h3
            class="meal-item__title"
          >
            Barbecue Burger
          </h3>
          <div
            class="meal-item__description"
          >
            American, raw, meaty
          </div>
          <div
            class="meal-item__price"
          >
            £12.99
          </div>
        </div>
        <form
          class="meal-item__add"
          name="Add meal item"
        >
          <div
            class="form__group"
          >
            <label
              class="form__label"
              for=":r6:"
            >
              Amount
            </label>
            <input
              class="form__input "
              id=":r6:"
              type="number"
              value="1"
            />
          </div>
          <button
            class="btn btn--primary btn--long "
          >
            + Add
          </button>
        </form>
      </div>
    </li>
    <li>
      <div
        class="meal-item"
      >
        <div
          class="meal-item__details"
        >
          <h3
            class="meal-item__title"
          >
            Green Bowl
          </h3>
          <div
            class="meal-item__description"
          >
            Healthy...and green...
          </div>
          <div
            class="meal-item__price"
          >
            £18.99
          </div>
        </div>
        <form
          class="meal-item__add"
          name="Add meal item"
        >
          <div
            class="form__group"
          >
            <label
              class="form__label"
              for=":r7:"
            >
              Amount
            </label>
            <input
              class="form__input "
              id=":r7:"
              type="number"
              value="1"
            />
          </div>
          <button
            class="btn btn--primary btn--long "
          >
            + Add
          </button>
        </form>
      </div>
    </li>
  </ul>
</section>
`;
